crates/sui-swarm/src/memory/node.rs

```diff
@@ -18,7 +19,7 @@ use super::container::Container;
 /// explicitly stopped by calling [`Node::stop`]) by simply dropping that Node's runtime.
 #[derive(Debug)]
 pub struct Node {
-    container: Option<Container>,
+    container: Mutex<Option<Container>>,
```

Comment:
Nothing wrong with that, but looking on it thought that maybe the [ArcSwapOption](https://docs.rs/arc-swap/latest/arc_swap/type.ArcSwapOption.html) could be an alternative here

Reply:
Cool! - I think Mutex is probably more readable for most purposes in this case so I will leave it as is unless you se a strong argument to change it.